packages/client-config/src/client-config-contributor/client_config_contributor_v1.ts

```diff
@@ -58,9 +72,181 @@ export class VersionContributorV1 implements ClientConfigContributor {
+        require_numbers: false,
+        require_symbols: false,
+        require_uppercase: false,
+      };
+      if (authOutput.payload.passwordPolicyMinLength) {
+        authClientConfig.auth.password_policy.min_length = Number.parseInt(
+          authOutput.payload.passwordPolicyMinLength
+        );
+      }
+      if (authOutput.payload.passwordPolicyRequirements) {
+        const requirements = JSON.parse(
+          authOutput.payload.passwordPolicyRequirements
+        ) as string[];
+        for (const requirement of requirements) {
+          switch (requirement) {
+            case 'REQUIRES_NUMBERS':
+              authClientConfig.auth.password_policy.require_numbers = true;
+              break;
+            case 'REQUIRES_LOWERCASE':
+              authClientConfig.auth.password_policy.require_lowercase = true;
+              break;
+            case 'REQUIRES_UPPERCASE':
+              authClientConfig.auth.password_policy.require_uppercase = true;
+              break;
+            case 'REQUIRES_SYMBOLS':
+              authClientConfig.auth.password_policy.require_symbols = true;
+              break;
+          }
+        }
+      }
+    }
+
+    // OAuth settings are present if both oauthRedirectSignIn and oauthRedirectSignOut are.
+    if (
+      authOutput.payload.oauthRedirectSignIn &&
+      authOutput.payload.oauthRedirectSignOut
+    ) {
+      let socialProviders = authOutput.payload.socialProviders
+        ? JSON.parse(authOutput.payload.socialProviders)
+        : [];
+      if (Array.isArray(socialProviders)) {
+        socialProviders = socialProviders.filter(this.isValidIdentityProvider);
+      }
+      authClientConfig.auth.oauth = {
+        identity_providers: socialProviders,
+        redirect_sign_in_uri: authOutput.payload.oauthRedirectSignIn.split(','),
+        redirect_sign_out_uri:
+          authOutput.payload.oauthRedirectSignOut.split(','),
+        response_type: authOutput.payload.oauthResponseType as 'code' | 'token',
+        scopes: authOutput.payload.oauthScope
+          ? JSON.parse(authOutput.payload.oauthScope)
+          : [],
+        domain: authOutput.payload.oauthCognitoDomain ?? '',
+      };
+    }
+
+    if (authOutput.payload.allowUnauthenticatedIdentities) {
+      authClientConfig.auth.unauthenticated_identities_enabled =
+        authOutput.payload.allowUnauthenticatedIdentities === 'true';
+    }
+
+    return authClientConfig;
+  };
+
+  // Define a type guard function to check if a value is a valid IdentityProvider
+  isValidIdentityProvider = (identityProvider: string): boolean => {
+    return [
+      'GOOGLE',
+      'FACEBOOK',
+      'LOGIN_WITH_AMAZON',
+      'SIGN_IN_WITH_APPLE',
+    ].includes(identityProvider);
+  };
+}
+
+/**
+ * Translator for the Auth portion of ClientConfig in V1.2
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export class AuthClientConfigContributorV1_1
+  implements ClientConfigContributor
+{
```

Comment:
Do we need to create a new contributor here? If the new field is optional, won't updating the existing one would work for generating both new and old configs?

Reply:
I followed the same pattern when we added storage access paths to outputs, the new field `paths` is optional but we still created a new contributor. This also minimizes the changes of us breaking existing versions.

Reply:
But we are duplicating quite a bit of code. Can we assert from tests that we won't break existing versions?